python/neuroglancer/ingest/tasks.py

```diff
@@ -347,6 +508,135 @@ def _encode(self, chunk, encoding):
     def _delete_data(self):
         self._datablob.delete()
 
+class HyperSquareTask(object):
+    def __init__(self, chunk_path=None, chunk_encoding=None, version=None, info_path=None, fromjson=None, _id=None):
+        self._id =  None
+        self.chunk_path = chunk_path
+        self.chunk_encoding = chunk_encoding
+        self.version = version
+        self.info_path = info_path
+        self.tag = 'hypersquare'
+        if fromjson:
+            self.payloadBase64 = fromjson
+            self._id = _id
+
+    @property
+    def payloadBase64(self):
+        payload = json.dumps({
+            'chunk_path': self.chunk_path,
+            'chunk_encoding': self.chunk_encoding,
+            'version': self.version,
+            'info_path': self.info_path
+        })
+        return base64.b64encode(payload)
+    
+    @payloadBase64.setter
+    def payloadBase64(self, payload):
+        decoded_string =  base64.b64decode(payload).encode('ascii')
+        d = json.loads(decoded_string)
+        self.chunk_path = d['chunk_path']
+        self.chunk_encoding = d['chunk_encoding']
+        self.version = d['version']
+        self.info_path = d['info_path']
+
+    def __repr__(self):
+        return "HyperSquareTask(chunk_path='{}, chunk_encoding='{}', version='{}', info_path='{}')".format(
+            self.chunk_path, self.chunk_encoding, self.version, self.info_path)
+
+    def execute(self):
+        self._parse_info_path()
+        self._parse_chunk_path()
+        self._storage = Storage(self._dataset_name, self._layer_name, compress=False)
+        self._download_metadata()
+        self._download_data()
+        self._upload_chunk()
+
+    def _parse_chunk_path(self):
+        if 'segmentation' in self._layer_name: 
+            match = re.match(r'^gs://(.*)/(.*)/segmentation.lzma', self.chunk_path)
+            self._bucket_name, self._chunk_folder = match.groups()
+        elif 'image' in self._layer_name:
+            match = re.match(r'^gs://(.*)/(.*)/jpg/0\.jpg', self.chunk_path)
+            self._bucket_name, self._chunk_folder = match.groups()
+        else:
+            return NotImplementedError("Don't know how process this layer")
+
+
+    def _parse_info_path(self):
+        match = re.match(r'^.*/([^//]+)/([^//]+)/info$', self.info_path)
+        self._dataset_name, self._layer_name = match.groups()
+
+    def _download_metadata(self):
+        self._bucket = self._storage._client.get_bucket(self._bucket_name)
+        metadata = self._bucket.get_blob(
+            '{}/metadata-fixed.json'.format(self._chunk_folder)) \
+        .download_as_string()
+        self._metadata = json.loads(metadata)
+        
+    def _download_data(self):
+        if 'segmentation' in self._layer_name: 
```

Comment:
1) Where is `self._layer_name` assigned to? Same for `self._dataset_name`
2) Is _layer_name a tuple? If not, using `in` is misleading and you should opt for `==`